### ISSUE TITLE
Lists Styles System

### ISSUE THREAD

**Issue by georgeolaru** [pixelgrade/rosa2#266]
A _list of items_ is a fundamental practice on every page content as it's used to organize information for users. There are a large variety of list styles being used: from the basic bulleted list of paragraphs to the checkmark list for pricing table characteristics.

---

**Styling trap**: Doing the research on how other tools are providing customizations for the lists, I often notice that there are numerous options available (eg. solid or dashed dividers, icons choices, borders to simulate a table) and most of them _doesn’t signify any purpose_ of the user. An example of purpose would be that the ordered list could be used to describe a sequence between list items like a food recipe.

[Here](https://www.are.na/george-olaru/lists-styling) I will cover some real-life examples that we can examine to identify _the intention_ and use them as references.

## The Lists Styles System
To add another stone to our *design system*, I suggest creating a *list system* which is intentional and purposeful. 

![Screenshot 2021-06-23 at 10-46](https://user-images.githubusercontent.com/1632775/123057143-4a892800-d410-11eb-911b-93d89e719613.jpg)

Some guidelines below:
- in addition to the default list styles (bullet and numbered) we will add a *no-bullet* style for general purposes and an *icon variation* for bullets to highlight certain features (eg. checkmarks for pricing tables)
- we will use the “Color Signal” component to pursue the user in deciding the importance of that list in a certain context: a low signal will make the list less prominent on a page.
- a way to increase or decrease the connection between items (eg. horizontal borders to separate them even more, vertical connectors to bring them together in a timeline style)

## Resources 
- [Figma Mockups ↗](https://www.figma.com/file/LkOxQKLMqMpZwBJQiTmmuiYK/Plugin-Nova-Blocks?node-id=3333%3A0)
- Siemens MindSphere Design System → [List Patterns](https://design.mindsphere.io/patterns/list.html)
- Google Material Design → [Lists](https://material.io/components/lists)
- GOV.UK Design System → [Lists](https://design-system.service.gov.uk/styles/typography/#lists)

**Comment by georgeolaru** [comment]
Good job so far @madalingorbanescu! I took a fresh look, and below is my review:

- [ ] **Markers** do not align vertically with the line of text. I think we need to find a way to position them dynamically and independent of the font size (eg. body text or headings with various font sizes)
- [ ] **Horizontal divider** should be limited by the list width on the left side and should not be constrained by the text content
- [ ] Within **Editor** I noticed that the list styles affect other blocks (eg. I notice some bullets when selecting the variations for the Columns block)
- [ ] Ordered list
   - I noticed that the counter doesn't take into consideration the `Start value` and `Reverse list numbering` block options
- [ ] List indentation
   - When the Checkmark is used for an indented list, the icon becomes smaller and shifts off the center
   - Adjust vertical margins of the indented `<ul>` so that the list is visually linked to the parent list, not the list below. For this, I suggest removing the space between list items of an indented list (or at least halved), so the top margin of the indented `<ul>` always remain larger than the spacing of its `<li>` (see [screenshot](https://user-images.githubusercontent.com/1632775/123537429-17a8a200-d738-11eb-88b6-506e4d867d24.jpg))

Here is a recording of the content that I used for testing:

![Screenshot 2021-06-27 at 10-55](https://user-images.githubusercontent.com/1632775/123537046-4291f680-d736-11eb-8c19-713b85d0aff8.gif)

And here is a gist with the sample content:
https://gist.github.com/georgeolaru/955ae4d85ac812528e46e410ffc65be3
